beets/importer.py

```diff
@@ -520,6 +523,34 @@ def remove_duplicates(self, lib):
                 util.prune_dirs(os.path.dirname(item.path),
                                 lib.directory)
 
+    def merge_duplicates(self, lib):
+        """Combine an import with an existing album.
+
+        Used after resolve_duplicates determines that there are
+        duplicates and the user selects merge.  Both sets of items have
+        already been added to the db; so this function combines them
+        into one list, zeroes out their ids, removes them from the db,
+        and re-adds them as a single album.
+        """
+
+        # Tag all with the same album id
+        existing_album = self.find_duplicates(lib)[0]
+        for item in self.items:
+            item.album_id = existing_album.id
+
+        # Must set album_id before setting choice_flag
+        # choice_flag used for self.import_items() in other methods
+        self.choice_flag = action.ASIS
+
+        for item in existing_album.items():
+            self.items.append(item)
+        self.align_album_level_fields()
+
+        # Set ids to None so add() "add"s rather than "store"s in lib.add_album
+        for item in self.items:
+            item.id = None
```

Comment:
Can you explain a little bit more about why this is necessary? Why should we re-add items that are already in the library, just associated with a different album?